pkg/api/job_runs.go

```diff
@@ -81,3 +86,182 @@ func JobsRunsReportFromDB(dbc *db.DB, filterOpts *filter.FilterOptions, release
 		Page:      pagination.Page,
 	}, res.Error
 }
+
+// JobRunAnalysis checks the test failures and linked bugs for a job run, and reports back an estimated
+// risk level for each failed test, and the job run overall.
+func JobRunAnalysis(dbc *db.DB, jobRunID int64) (apitype.ProwJobRunFailureAnalysis, error) {
+
+	jobRun := &models.ProwJobRun{}
+	// Load the ProwJobRun, ProwJob, and failed tests:
+	// TODO: we may want to expand to analyzing flakes here in the future
+	res := dbc.DB.Joins("ProwJob").
+		Preload("ProwJob.Bugs", "status != 'Closed' and status != 'Verified'").
+		Preload("Tests", "status = 12").
+		Preload("Tests.Test").Preload("Tests.Test.Bugs").
+		Preload("Tests.Suite").First(jobRun, jobRunID)
+	if res.Error != nil {
+		return apitype.ProwJobRunFailureAnalysis{}, res.Error
+	}
+
+	// If this job is a Presubmit, compare to test results from master, not presubmits, which may perform
+	// worse due to dev code that hasn't merged. We do not presently track presubmits on branches other than
+	// master, so it should be safe to assume the latest compareRelease in the db.
+	compareRelease := jobRun.ProwJob.Release
+	if compareRelease == "Presubmits" {
+		// Get latest release from the DB:
+		ar, err := query.ReleasesFromDB(dbc)
+		if err != nil {
+			return apitype.ProwJobRunFailureAnalysis{}, err
+		}
+		if len(ar) == 0 {
+			return apitype.ProwJobRunFailureAnalysis{}, fmt.Errorf("no releases found in db")
+		}
+
+		compareRelease = ar[0].Release
```

Comment:
Any issues here when we are setting up a new release but haven't switched yet>

Reply:
Excellent question, that did not occur to me. 

I think the way that works (IIRC), dptp tends to make release-4.13 git branches in all repos first, and they're automatically kept in sync with master. New jobs get created with 4.13 in the name, but sippy won't start scraping them until we explicitly tell it about 4.13. During that time, presubmits are technically 4.13 if it's against a master branch, but we'd be comparing to test pass rates for 4.12. I think that's probably actually ok and the best we could hope for. Seems like it will be ok but it's not guaranteed.